magic_the_gathering/actions/tap.py

```diff
@@ -0,0 +1,37 @@
+from typing import List
+
+from magic_the_gathering.actions.base import Action
+from magic_the_gathering.game_state import GameState, ZonePosition
+
+
+class TapAction(Action):
+    @classmethod
+    def list_possible_actions(cls, game_state: GameState) -> List[Action]:
+        possible_actions = []
+        for player_index, player in enumerate(game_state.players):
+            if player.is_alive:
+                player_board = game_state.zones[ZonePosition.BOARD][player_index]
+                for card_uuid, card in player_board.items():
+                    if not card.state.is_tapped:
+                        possible_actions.append(
+                            cls(
+                                owner=f"Player {player_index}",
+                                player_index=player_index,
+                                card_uuid=card_uuid,
+                            )
+                        )
+        return possible_actions
+
+    def __init__(self, owner: str, player_index: int, card_uuid: str):
+        super().__init__(owner=owner)
+        self.player_index = player_index
+        self.card_uuid = card_uuid
+
+    def _execute(self, game_state: GameState) -> GameState:
+        player_board = game_state.zones[ZonePosition.BOARD][self.player_index]
+        assert self.card_uuid in player_board.keys()
+        player_card = player_board[self.card_uuid]
+        # FIXME: Increase mana pool if land
```

Comment:
Yes exactly, it is the activated ability that produces mana, however we currently don't support abilities at all, this is why I marked it as a FIXME :sweat_smile: Yesterday I was hesitating, like maybe for now we could create mana as the result of tapping lands, otherwise we need to start thinking about the logic for abilities. I don't know which way is best

Reply:
> split cards

WTF? One card actually being two cards? :grin:

Reply:
Oh sure we could have land tapping = mana produced for now